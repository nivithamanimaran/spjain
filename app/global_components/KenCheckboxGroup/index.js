import { useTheme } from '@material-ui/core';
import React, { useEffect, useState } from 'react';
import KenCheckbox from '../KenCheckbox';

const KenCheckboxGroup = props => {
  const theme = useTheme();
  const { value, options = [], onChange } = props;

  const [checkedArray, setCheckedArray] = useState([]);

  useEffect(() => {
    if (!Array.isArray(value)) {
      setCheckedArray([]);
    } else {
      setCheckedArray(value);
    }
  }, [value]);

  const handleOnChange = e => {
    if (checkedArray.includes(e.target.name)) {
      onChange(checkedArray.filter(d => d !== e.target.name));
    } else {
      onChange([...checkedArray, e.target.name]);
    }
  };

  return (
    <div data-testid="checkbox-group">
      {options.length
        ? options.map(option => {
            return (
              <KenCheckbox
                key={option.value}
                disableRipple={true}
                name={option.value}
                value={checkedArray.includes(option.value)}
                label={option.label}
                onChange={handleOnChange}
                labelProps={{
                  style: {
                    fontSize: 12,
                    color: theme.palette.KenColors.neutral400,
                  },
                }}
              />
            );
          })
        : null}
    </div>
  );
};

export default KenCheckboxGroup;
