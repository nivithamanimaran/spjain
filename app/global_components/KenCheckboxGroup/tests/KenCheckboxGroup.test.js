import React from 'react';
import { render, cleanup, fireEvent } from 'react-testing-library';
import '@testing-library/jest-dom/extend-expect';
import ThemeProvider from '../../../utils/themeProvider';
import { BrowserRouter as Router } from 'react-router-dom';
import KenCheckboxGroup from '..';

afterEach(() => {
  cleanup();
});

const options = [
  {
    label: 'item 1',
    value: 'item1',
  },
  {
    label: 'item 2',
    value: 'item2',
  },
  {
    label: 'item 3',
    value: 'item3',
  },
];

describe('Checkbox group test', () => {
  it('should render and match the snapshot and only check the one which is present in value array', () => {
    const onChange = jest.fn();

    const { container, getByTestId } = render(
      <ThemeProvider>
        <Router>
          <KenCheckboxGroup
            options={options}
            value={['item1']}
            onChange={onChange}
          />
        </Router>
      </ThemeProvider>
    );

    expect(getByTestId('checkbox-group')).toMatchSnapshot();
    expect(container.querySelector('input[name="item1"]')).toBeChecked();
    expect(container.querySelector('input[name="item2"]')).not.toBeChecked();
    expect(container.querySelector('input[name="item3"]')).not.toBeChecked();
  });

  it('onChange should be called with updated list of values', () => {
    const onChange = jest.fn();
    const { container, getByTestId } = render(
      <ThemeProvider>
        <Router>
          <KenCheckboxGroup
            options={options}
            value={['item1']}
            onChange={onChange}
          />
        </Router>
      </ThemeProvider>
    );

    expect(getByTestId('checkbox-group')).toMatchSnapshot();
    fireEvent.click(container.querySelector('input[name="item2"]'));
    expect(onChange).toHaveBeenCalledTimes(1);
    expect(onChange).toHaveBeenCalledWith(['item1', 'item2']);
  });
});
