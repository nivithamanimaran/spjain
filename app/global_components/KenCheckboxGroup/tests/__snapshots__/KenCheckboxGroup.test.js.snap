// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox group test onChange should be called with updated list of values 1`] = `
<div
  data-testid="checkbox-group"
>
  <div
    class="MuiFormGroup-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
      style="flex-wrap: nowrap;"
    >
      <label
        class="MuiFormControlLabel-root"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-13 MuiCheckbox-root Component-root-11 makeStyles-transparentBg-10 PrivateSwitchBase-checked-14 Mui-checked Component-checked-12"
          component-root="[object Object]"
          style="margin-right: -20px;"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              class="PrivateSwitchBase-input-16"
              data-indeterminate="false"
              name="item1"
              type="checkbox"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
              />
            </svg>
          </span>
        </span>
        <span
          class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        />
      </label>
      <p
        class="MuiTypography-root MuiTypography-body1"
        style="font-size: 12px; color: rgb(80, 95, 121);"
      >
         
        item 1
         
      </p>
    </div>
  </div>
  <div
    class="MuiFormGroup-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
      style="flex-wrap: nowrap;"
    >
      <label
        class="MuiFormControlLabel-root"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-13 MuiCheckbox-root Component-root-11 makeStyles-transparentBg-10"
          component-root="[object Object]"
          style="margin-right: -20px;"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              class="PrivateSwitchBase-input-16"
              data-indeterminate="false"
              name="item2"
              type="checkbox"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </span>
        </span>
        <span
          class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        />
      </label>
      <p
        class="MuiTypography-root MuiTypography-body1"
        style="font-size: 12px; color: rgb(80, 95, 121);"
      >
         
        item 2
         
      </p>
    </div>
  </div>
  <div
    class="MuiFormGroup-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
      style="flex-wrap: nowrap;"
    >
      <label
        class="MuiFormControlLabel-root"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-13 MuiCheckbox-root Component-root-11 makeStyles-transparentBg-10"
          component-root="[object Object]"
          style="margin-right: -20px;"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              class="PrivateSwitchBase-input-16"
              data-indeterminate="false"
              name="item3"
              type="checkbox"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </span>
        </span>
        <span
          class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        />
      </label>
      <p
        class="MuiTypography-root MuiTypography-body1"
        style="font-size: 12px; color: rgb(80, 95, 121);"
      >
         
        item 3
         
      </p>
    </div>
  </div>
</div>
`;

exports[`Checkbox group test should render and match the snapshot and only check the one which is present in value array 1`] = `
<div
  data-testid="checkbox-group"
>
  <div
    class="MuiFormGroup-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
      style="flex-wrap: nowrap;"
    >
      <label
        class="MuiFormControlLabel-root"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiCheckbox-root Component-root-3 makeStyles-transparentBg-2 PrivateSwitchBase-checked-6 Mui-checked Component-checked-4"
          component-root="[object Object]"
          style="margin-right: -20px;"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              class="PrivateSwitchBase-input-8"
              data-indeterminate="false"
              name="item1"
              type="checkbox"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
              />
            </svg>
          </span>
        </span>
        <span
          class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        />
      </label>
      <p
        class="MuiTypography-root MuiTypography-body1"
        style="font-size: 12px; color: rgb(80, 95, 121);"
      >
         
        item 1
         
      </p>
    </div>
  </div>
  <div
    class="MuiFormGroup-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
      style="flex-wrap: nowrap;"
    >
      <label
        class="MuiFormControlLabel-root"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiCheckbox-root Component-root-3 makeStyles-transparentBg-2"
          component-root="[object Object]"
          style="margin-right: -20px;"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              class="PrivateSwitchBase-input-8"
              data-indeterminate="false"
              name="item2"
              type="checkbox"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </span>
        </span>
        <span
          class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        />
      </label>
      <p
        class="MuiTypography-root MuiTypography-body1"
        style="font-size: 12px; color: rgb(80, 95, 121);"
      >
         
        item 2
         
      </p>
    </div>
  </div>
  <div
    class="MuiFormGroup-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
      style="flex-wrap: nowrap;"
    >
      <label
        class="MuiFormControlLabel-root"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiCheckbox-root Component-root-3 makeStyles-transparentBg-2"
          component-root="[object Object]"
          style="margin-right: -20px;"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              class="PrivateSwitchBase-input-8"
              data-indeterminate="false"
              name="item3"
              type="checkbox"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </span>
        </span>
        <span
          class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        />
      </label>
      <p
        class="MuiTypography-root MuiTypography-body1"
        style="font-size: 12px; color: rgb(80, 95, 121);"
      >
         
        item 3
         
      </p>
    </div>
  </div>
</div>
`;
